parse_date.re

```diff
@@ -2200,7 +2206,12 @@ timelib_time *timelib_parse_from_format(char *format, char *string, size_t len,
 					break;
 
 				default:
-					add_pbf_error(s, TIMELIB_ERR_DATA_MISSING, "Data missing", string, ptr);
+					int len = strlen(format) + strlen(string) + 80;
+					char *msg = timelib_calloc(sizeof(char), len);
+					sprintf(msg, "DateTime format (%s) indicates that data is missing from "
```

Comment:
Same here, please calculate the format size better, and don't use PHP-isms like "DateTime".